### PR TITLE
Implement `API Tokens` mutations

### PR DESCRIPTION
**Pull request by vyzaldysanchez**
Closes: https://app.shortcut.com/chainlinklabs/story/20933/implement-api-token-mutations

This aims to allow consumers to create/delete API tokens via GQL.

### CODE REVIEW

File: core/web/schema/type/api_token.graphql [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+union CreateAPITokenPayload = CreateAPITokenSuccess | InputErrors
+
+input DeleteAPITokenInput {
+    password: String!
+}
+
+type DeleteAPITokenResult {
+    accessKey: String!
+    # Not returning the secret since it is hashed in the DB
```
Comment (jkongie) [inline core/web/schema/type/api_token.graphql:22]: Yes that's correct you don't need to return the API token. You can remove this comment

**Review by jkongie — CHANGES_REQUESTED** [review]
Looks good. Just remove a comment

**Review by jkongie — APPROVED** [review]

File: core/sessions/orm.go [hunk 2]
```diff
@@ -224,6 +225,17 @@ func (o *orm) SetPassword(user *User, newPassword string) error {
 	return o.db.Get(user, sql, hashedPassword, user.Email)
 }
 
+func (o *orm) GenerateAuthToken(user *User) (*auth.Token, error) {
```
Comment (makramkd) [inline core/sessions/orm.go:228]: Minor: the name of this method suggests that it's pure and not setting anything in the DB (i.e it's just generating a token and not updating a row in the database). Why not call `SetAuthToken(user, auth.NewToken())` instead?

**Review by makramkd — COMMENTED** [review]
Minor comment, LGTM otherwise

File: core/sessions/orm.go (hunk 2 above)
Comment (vyzaldysanchez) [reply core/sessions/orm.go:228]: Difficult to mock and control which token I would want to actually set in the DB without relying on randomness when generating the new token, hence why I created an ORM method that encapsulates the random part in order to mock and test properly...

Comment (makramkd) [reply core/sessions/orm.go:228]: That's a good point about mocking and testing. Maybe call this `CreateAndSetAuthToken`? Kind of a mouthful but clearer in it's intentions.

**Review by makramkd — DISMISSED** [review]
🚀

**Review by jkongie — DISMISSED** [review]

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 3c3e455e98a01db4fe4db5f015dede59335a5e0d) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Review by jkongie — APPROVED** [review]

**Pull request merged**